deepchem/models/torch_models/tests/test_progressive_multitask.py

```diff
@@ -109,3 +110,125 @@ def to_torch_param(weights):
 
     assert np.allclose(output, torch_out,
                        atol=1e-4), "Predictions are not close"
+
+
+@pytest.mark.torch
+def test_progressivemultitask_regression_overfit():
+    """Test that the model can overfit simple regression datasets."""
+    n_samples = 10
+    n_features = 5
+    n_tasks = 3
+
+    np.random.seed(123)
+    torch.manual_seed(123)
+    X = np.random.rand(n_samples, n_features)
+    y = np.random.rand(n_samples, n_tasks).astype(np.float32)
+    dataset = dc.data.NumpyDataset(X, y)
+
+    regression_metric = dc.metrics.Metric(dc.metrics.mean_squared_error,
+                                          task_averager=np.mean,
+                                          mode="regression")
+
+    model = dc.models.torch_models.ProgressiveMultitaskModel(
+        n_tasks,
+        n_features,
+        layer_sizes=[128, 256],
+        dropouts=0.2,
+        alpha_init_stddevs=0.02,
+        weight_init_stddevs=0.02,
+        bias_init_consts=0.0,
+        mode='regression',
+    )
+
+    model.fit(dataset, nb_epoch=200)
+
+    scores = model.evaluate(dataset, [regression_metric])
+    assert scores[regression_metric.name] < 0.05, "Failed to overfit"
+
+
+@pytest.mark.torch
+def test_progressivemultitask_classification_overfit():
```

Comment:
I think for classification mode one more unit test can be added wherein the results of both Tensorflow and PyTorch be compared when same input data and same weights and biases of different layers involved in the model are given to both Tensorflow and PyTorch implementation.

Reply:
The `test_progressivemultitask_forward()` unit test initially had something similar where the existing TensorFlow model was trained and its weights ported to the equivalent layers in the PyTorch model. A forward pass was done on both the models then with sample data and checked if they were the same. Dr. Bharat asked to remove the tf implementation since we would be removing it in the future and save the weights and outputs of the tf model to a file instead from where it will be loaded onto the PyTorch model. 

I feel that the existing test for forward pass is enough for comparing the models and will be repeating most of it if we add another test for classification too.

Reply:
I see. Sorry my bad. You have already added that test.

Reply:
@rbharath Sir this looks good to me.